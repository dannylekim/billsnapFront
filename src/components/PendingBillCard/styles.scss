@import "../../variables";

div.active__bill {
  background-color: rgba(241, 241, 241, 0.71);
}

.date {
  font-weight: 500;
  color: rgba(0, 0, 0, 0.56);
  font-size: 18px;
}

div.bill__items {
  text-align: left;
  display: flex;
  justify-content: space-between;
}

.bill__name {
  font-family: $primary-font;
  font-style: normal;
  font-weight: bold;
  font-size: 20px;
  color: $primary-text-color;
}

.bill-icon {
  margin-top: 0.5rem;
}

.separate-right {
  margin-right: 1em;
}

.split__info {
  display: flex;
  justify-content: space-between;
}

.money__info {
  font-family: $primary-font;
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  color: $primary-text-color;
}

.bill-balance {
  color: #ffb400;
}

.responsible {
  font-weight: bold;
  font-family: $primary-font;
  color: $tertiary-text-color;
}
