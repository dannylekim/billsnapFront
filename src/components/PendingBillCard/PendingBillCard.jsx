import React from "react";
import "./styles.scss";
import {MdThumbDown, MdThumbUp} from "react-icons/md";
import BillIcon from "../BillIcon";

export const PendingBillCard = ({
  acceptBillInvitationHandler,
  declineBillInvitationHandler,
  filterDateHandler,
  bill,
  activeBillId,
  activeBillHandler,
}) => {
  return (
    <div
      onClick={activeBillHandler}
      className={
        activeBillId === bill.id
          ? "active__bill card-body"
          : "non__active__bill card-body"
      }
    >
      <div className="split__info">
        <div />
        <div className="date">{filterDateHandler(bill.created)}</div>
      </div>
      <div className="bill__items">
        <div className="split__info">
          <div className="separate-right bill-icon">
            <BillIcon category={bill.category} />
          </div>
          <div className="bill__name">
            <div>{bill.name}</div>
            <div className="responsible">
              Split by : {bill.responsible.firstName}
            </div>
          </div>
        </div>
        <div className="money__info">
          <div className="split__info">
            <div> Bill Total </div>
            <div> {parseFloat(bill.balance).toFixed(2)} </div>
          </div>
          <div className="split__info">
            <div className="separate-right">You will owe </div>
            <div className="bill-balance">{bill.amountOwed}</div>
          </div>

          <div className="split__info">
            <div> Accept? </div>
            <div>
              <MdThumbUp
                className="icon__button accept"
                size={20}
                onClick={acceptBillInvitationHandler}
              />
            </div>
            <div>
              <MdThumbDown
                className="icon__button decline"
                size={20}
                onClick={declineBillInvitationHandler}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
