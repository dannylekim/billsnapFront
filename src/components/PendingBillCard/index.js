import PendingBillCard from "./PendingBillCard";

export default PendingBillCard;
