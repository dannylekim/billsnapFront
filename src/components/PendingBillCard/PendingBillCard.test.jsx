import React from "react";
import {PendingBillCard} from "./PendingBillCard";

describe("PendingBillCard", () => {

  let wrapper;

  describe("render", () => {
    const bill = {
      id: 0,
      name: "Test Bill",
      responsible: {
        firstName: "Test User",
      },
      status: "IN_PROGRESS",
      category: "string",
      created: "2020-08-14T15:28:56.580Z",
      balance: 100,
      amountOwed: 50
    };

    describe("snapshots ðŸ“¸", () => {
      it("PendingBillCard should match snap shot", () => {
        matches(
          <PendingBillCard
            bill={bill}
            activeBillId={bill.id}
            acceptBillInvitationHandler={jest.fn()}
            declineBillInvitationHandler={jest.fn()}
            filterDateHandler={jest.fn()}
            getBillIconHandler={jest.fn()}
          />
        );
      });
    });

    describe("component ", () => {
      it("should not find an active bill", () => {
        wrapper = shallow(
            <PendingBillCard
                bill={bill}
                activeBillId={999}
                acceptBillInvitationHandler={jest.fn()}
                declineBillInvitationHandler={jest.fn()}
                filterDateHandler={jest.fn()}
                getBillIconHandler={jest.fn()}
            />
        );
        expect(wrapper.find(".non__active__bill")).toHaveLength(1);
      });

      it("should find an active bill", () => {
        wrapper = shallow(
            <PendingBillCard
                bill={bill}
                activeBillId={bill.id}
                acceptBillInvitationHandler={jest.fn()}
                declineBillInvitationHandler={jest.fn()}
                filterDateHandler={jest.fn()}
                getBillIconHandler={jest.fn()}
            />
        );
        expect(wrapper.find(".active__bill")).toHaveLength(1);
      });
    });
  });
});
