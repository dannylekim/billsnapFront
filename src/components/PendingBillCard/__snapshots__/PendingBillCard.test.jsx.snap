// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingBillCard render snapshots ðŸ“¸ PendingBillCard should match snap shot 1`] = `
<div
  className="active__bill card-body"
>
  <div
    className="split__info"
  >
    <div />
    <div
      className="date"
    />
  </div>
  <div
    className="bill__items"
  >
    <div
      className="split__info"
    >
      <div
        className="separate-right bill-icon"
      >
        <svg
          color="rgba(0, 0, 0, 0.96)"
          fill="currentColor"
          height={24}
          size={24}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.96)",
            }
          }
          viewBox="0 0 384 512"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M202.021 0C122.202 0 70.503 32.703 29.914 91.026c-7.363 10.58-5.093 25.086 5.178 32.874l43.138 32.709c10.373 7.865 25.132 6.026 33.253-4.148 25.049-31.381 43.63-49.449 82.757-49.449 30.764 0 68.816 19.799 68.816 49.631 0 22.552-18.617 34.134-48.993 51.164-35.423 19.86-82.299 44.576-82.299 106.405V320c0 13.255 10.745 24 24 24h72.471c13.255 0 24-10.745 24-24v-5.773c0-42.86 125.268-44.645 125.268-160.627C377.504 66.256 286.902 0 202.021 0zM192 373.459c-38.196 0-69.271 31.075-69.271 69.271 0 38.195 31.075 69.27 69.271 69.27s69.271-31.075 69.271-69.271-31.075-69.27-69.271-69.27z"
          />
        </svg>
      </div>
      <div
        className="bill__name"
      >
        <div>
          Test Bill
        </div>
        <div
          className="responsible"
        >
          Split by : 
          Test User
        </div>
      </div>
    </div>
    <div
      className="money__info"
    >
      <div
        className="split__info"
      >
        <div>
           Bill Total 
        </div>
        <div>
           
          100.00
           
        </div>
      </div>
      <div
        className="split__info"
      >
        <div
          className="separate-right"
        >
          You will owe 
        </div>
        <div
          className="bill-balance"
        >
          50
        </div>
      </div>
      <div
        className="split__info"
      >
        <div>
           Accept? 
        </div>
        <div>
          <svg
            className="icon__button accept"
            fill="currentColor"
            height={20}
            onClick={[MockFunction]}
            size={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-1.91l-.01-.01L23 10z"
            />
          </svg>
        </div>
        <div>
          <svg
            className="icon__button decline"
            fill="currentColor"
            height={20}
            onClick={[MockFunction]}
            size={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15 3H6c-.83 0-1.54.5-1.84 1.22l-3.02 7.05c-.09.23-.14.47-.14.73v1.91l.01.01L1 14c0 1.1.9 2 2 2h6.31l-.95 4.57-.03.32c0 .41.17.79.44 1.06L9.83 23l6.59-6.59c.36-.36.58-.86.58-1.41V5c0-1.1-.9-2-2-2zm4 0v12h4V3h-4z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
